product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org.wso2.carbon.ei.scenario.test/AddChildToXMLTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *      http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.ei.scenario.test;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpResponse;
+import org.jaxen.JaxenException;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioConstants;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.wso2.carbon.esb.scenario.test.common.http.HTTPUtils;
+import org.wso2.carbon.esb.scenario.test.common.http.SOAPClient;
+import org.wso2.carbon.esb.scenario.test.common.utils.FileUtils;
+import org.wso2.carbon.esb.scenario.test.common.utils.XMLUtils;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class AddChildToXMLTest extends ScenarioTestBase {
+
+    private static final Log log = LogFactory.getLog(AddChildToXMLTest.class);
+    private String sourcesFilePath;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+        sourcesFilePath = testResourcesDir + File.separator + "source_files";
+    }
+
+    @Test(description = "1.6.1.1 - [SOAP > POX] enrich a inline content as a child")
+    public void testAddingInlineContentAsChild() throws IOException, XMLStreamException {
+        String url = getProxyServiceURLHttp("1_6_1_1_SoapToPoxMsgEnrichWithChildProxy");
+        String request = FileUtils.readFile(sourcesFilePath + File.separator + "request_1_6_1_1.xml");
```

Comment:
Can we have a util method for this?

Reply:
Added a util method is FileUtils.readFile(String pathToFile); I am using it here. :)